r/tests/testthat/test-dplyr-string-functions.R

```diff
@@ -467,6 +467,25 @@ test_that("strsplit and str_split", {
   )
 })
 
+test_that("str_to_lower, str_to_upper, and str_to_title", {
+  df <- tibble(x = c("1Foo1", " \tB a R\n", "!apACHe aRroW!"))
+  funcs <- c(str_to_lower, str_to_upper, str_to_title)
+  for (func in funcs) {
+    expect_dplyr_equal(
+      input %>%
+        transmute(x = func(x)) %>%
+        collect(),
+      df
+    )
+
+    funcname = as.character(substitute(func))
+    expect_error(
+      nse_funcs[[funcname]]("Apache Arrow", locale = "sp"),
+      "Providing a value for 'locale' other than the default ('en') is not supported by Arrow"
+    )
```

Comment:
We could probably get away with testing the error on one function and not wrapping them all in this for loop.